### PR TITLE
🐛(marsha) manage case where cloudfront private key is not defined

### LINKED ISSUE #180: marsha not installable anymore
## Bug Report

**Expected behavior/code**

When I run init playbook with marsha app I expect no failure

**Actual Behavior**

Creating masha secrets fails whit this message:

```
TASK [Actually create the secrets in OpenShift] **********************************************************************************************************************************************
changed: [local] => (item=apps/marsha/templates/app/secret.yml.j2)
failed: [local] (item=apps/marsha/templates/app/secret_cloudfront_private_key.yml.j2) => {"changed": false, "error": 422, "item": "apps/marsha/templates/app/secret_cloudfront_private_key.yml.j2", "msg": "Failed to create object: Secret \"marsha-sshkey-73fcc8eea30738a6b9647f12a9e7ac8f877a64fa\" is invalid: data[%s][ssh-privatekey]: Required value"}
changed: [local] => (item=apps/marsha/templates/postgresql/secret.yml.j2)
```

**Steps to Reproduce**
1. Update your master branch
2. Run `./bin/bootstrap -e "apps_filter=marsha`

**Possible Solution**

Something was removed I think when we worked on creating vaults.

### PR DESCRIPTION
**Pull request by lunika**
cloudfront private key is not required and can be not defined when you
deploy marsha. We have to manage this case and remove resources related
to this private key

## Purpose

Marsha is not installable anymore without defining cloudfront private key but this is not a requirement in marsha apps.

Fixes #180 

## Proposal

Manage case where the private key is not defined and not create resources linked to this secret key if not present.

### CODE REVIEW

**Review by jmaupetit — COMMENTED** [review]
I think we expect `DJANGO_CLOUDFRONT_PRIVATE_KEY` to be defined or not, not `DJANGO_CLOUDFRONT_PRIVATE_KEY_PATH`. Am I wrong?

**Comment by lunika** [comment]
Indeed I mixed two variables. It should be better now

File: apps/marsha/templates/app/dc.yml.j2 [hunk 1]
```diff
@@ -38,8 +38,10 @@ spec:
               value: "{{ marsha_postgresql_port }}"
             - name: DJANGO_ALLOWED_HOSTS
               value: "{{ marsha_host | blue_green_hosts }}"
+{% if DJANGO_CLOUDFRONT_PRIVATE_KEY_PATH is defined %}
```
Comment (jmaupetit) [inline apps/marsha/templates/app/dc.yml.j2]: I think `DJANGO_CLOUDFRONT_PRIVATE_KEY_PATH` is always defined (with a good default). Does it matter if it is defined while `DJANGO_CLOUDFRONT_PRIVATE_KEY` is not? If not, this block condition should be removed.

Comment (lunika) [reply apps/marsha/templates/app/dc.yml.j2]: :+1:

**Comment by lunika** [comment]
This PR is in fact a little bit hard to resolve because it introduces a new issue:

- It is not possible to have conditionnal template (empty due to a if block) with a variable defined in a vault because vault are not loaded in the `get_objects_for_app` task...

**Comment by lunika** [comment]
This PR is in fact a little bit hard to resolve because it introduces a new issue:

- It is not possible to have conditionnal template (empty due to a if block) with a variable defined in a vault because vault are not loaded in the `get_objects_for_app` task...

**Comment by lunika** [comment]
This PR is in fact a little bit hard to resolve because it introduces a new issue:

- It is not possible to have conditionnal template (empty due to a if block) with a variable defined in a vault because vault are not loaded in the `get_objects_for_app` task...

Should I load the vault variable in the `get_objects_for_app` task ?

**Comment by lunika** [comment]
This PR is in fact a little bit hard to resolve because it introduces a new issue:

- It is not possible to have conditionnal template (empty due to a if block) with a variable defined in a vault because vault are not loaded in the `get_objects_for_app` task...

Should I load the vault variable in the `get_objects_for_app` task ?

**Comment by lunika** [comment]
This PR is in fact a little bit hard to resolve because it introduces a new issue:

- It is not possible to have conditionnal template (empty due to a if block) with a variable defined in a vault because vault are not loaded in the `get_objects_for_app` task...

Should I load the vault variable in the `get_objects_for_app` task ?

A workaround only for this PR can be to use trashable env ans decide to exclude the ssh key if env_type is in trashable_env_types

**Comment by jmaupetit** [comment]
> Should I load the vault variable in the get_objects_for_app task ?

Yes, you can.

**Comment by lunika** [comment]
@jmaupetit modification made

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**